qt-models/yearlystatisticsmodel.cpp

```diff
@@ -225,4 +227,42 @@ void YearlyStatisticsModel::update_yearly_stats()
 		rootItem->children.append(item);
 		item->parent = rootItem.get();
 	}
+
+	/* Show the statistic sorted by dive depth */
+	if (stats.stats_by_depth != NULL && stats.stats_by_depth[0].is_trip == true &&
+			stats.stats_by_depth[0].selection_size) {
+		YearStatisticsItem *item = new YearStatisticsItem(stats.stats_by_depth[0]);
```

Comment:
Idle comment: Please don't take this as an example how memory management is done in C++ ☹
But it's consistent with the rest of the code, so OK.

Reply:
For my education and if you have free cycles, can you expand on why this is a bad example of C++ memory management?